src/main/java/meditracker/argument/ArgumentParser.java

```diff
@@ -59,6 +67,32 @@ public void checkForMissingRequiredArguments() throws ArgumentNotFoundException
         }
     }
 
+    // @@author wenenhoe-reused
+    // Reused from https://github.com/wenenhoe/ip with minor modifications
+    /**
+     * Checks if unknown argument flags are found in user input
+     *
+     * @param rawInputSplit List of raw input split by spaces
+     * @throws UnknownArgumentFoundException When unknown argument flags found in user input
+     */
+    public void checkForUnknownArguments(List<String> rawInputSplit) throws UnknownArgumentFoundException {
+        List<String> argumentFlags = argumentList.getArguments().stream()
+                .map(Argument::getFlag)
+                .collect(Collectors.toList());
+
+        String pattern = "^-.+$";
```

Comment:
It will be great if you could provide a description as to what this regex aims to match (either as a comment or a descriptive variable name)

Reply:
Resolved in 10ac337